### PR TITLE
docs: clean up and restructure plans/ documentation

### PR DESCRIPTION
**Pull request by tlmquintino**
## Summary

Collaborative cleanup of all markdown documents under `plans/`, cross-referenced against the full codebase for accuracy.

### New files
- **`plans/MOTIVATION.md`** — why Tensogram exists, demand evidence, adoption wedge, success criteria

### Restructured files
- **`plans/DESIGN.md`** — rewritten as clean design rationale (501→223 lines). Removed wire format details (covered by WIRE_FORMAT.md), outdated V1 metadata examples, process artifacts. Fixed API signatures (`decode_range`, `scan`, file API), corrected FFI function count (74→62), removed non-existent CLI `-d` flag.
- **`plans/PLAN.md`** — retitled "Iteration 0 Plan (Historical)" with banner pointing to living docs. Scope table annotated with current status.
- **`plans/DONE.md`** — cleaned to current state (421→152 lines). Removed historical timeline sections. Verified counts: 572 tests, 11 Rust examples, 8 Python examples, 62 FFI functions, 934-line C++ wrapper, 543-line header.
- **`plans/TEST.md`** — updated from initial plan to match actual test suite. All 6 compressors, mmap/async marked implemented, xarray tests added.
- **`plans/STYLE.md`** — expanded with per-language conventions (Rust, C FFI, C++, Python).
- **`plans/TODO.md`** — populated with decided features (reserved provenance fields, CI matrix, coverage target).
- **`CHANGELOG.md`** — expanded with all 4 releases (0.1.0–0.4.0) from GitHub.

### Minor fixes
- **`plans/WIRE_FORMAT.md`** — 5 typos fixed (Length, Header, description)
- **`plans/IDEAS.md`** — xarray marked as done
- **`CLAUDE.md`** — document references updated to include all 9 plans/ files

### Verification
- `cargo build` / `cargo fmt` / `cargo clippy` — clean
- `cargo test` — 257 Rust tests pass
- C++ build + ctest — 105 tests pass
- Python lint/format/pytest — 97 tests pass
- All 11 Rust examples run
- mdbook docs build

Docs Preview
https://sites.ecmwf.int/docs/dev-section/tensogram/pull-requests/PR-10

### CODE REVIEW

File: plans/WIRE_FORMAT.md [hunk 1]
```diff
@@ -94,7 +94,7 @@ Either a header or a footer metadata frame must always be present, ie Messages c
 
 ## Metadata Frame
 
-Irrespective of position, Hearder or Footer, the metadata frame for uniquely identifies the message.
+Irrespective of position, Header or Footer, the metadata frame uniquely identifies the message.
 Each metadata CBOR contains a mandatory sub-objects:
  - 'common' holds metadata that is common to all data objects in the message. this may contain internal namespaces for managing different vocabularies.
```
[the remark this answers is not in the document]
Comment (tlmquintino) [reply plans/WIRE_FORMAT.md]: Fixed in f27bb4c — rewrote to: "Whether in the Header or Footer, the metadata frame uniquely identifies the message. Each metadata CBOR block contains the following mandatory sub-objects:"

File: plans/TODO.md [hunk 2]
```diff
@@ -1,17 +1,33 @@
...
+
+- [ ] Document all error paths in docs/ (error handling reference page)
+
+## Builds
+
+- [ ] CI matrix for all three language test suites on every commit (partially done — Rust, Python, C++ run but GRIB tests gated on ecCodes availability)
+
+## Code Quality
+
+- [ ] Reach 95% test coverage (currently 90.5%)
```
[the remark this answers is not in the document]
Comment (tlmquintino) [reply plans/TODO.md]: Fixed in f27bb4c — changed to "line coverage" to match the metric.

File: plans/DONE.md [hunk 3]
```diff
@@ -1,421 +1,152 @@
...
-- 181 Rust tests
-- 95 Python tests
-- 103 C++ tests
-- 0 clippy warnings
-
-## C++ wrapper & build system (2026-04-04)
-
-### Header-only C++17 wrapper (`include/tensogram.hpp`)
-- `tensogram` namespace wrapping entire C API
+- Auto-generated `tensogram.h` (543 lines) via cbindgen
```
[the remark this answers is not in the document]
Comment (tlmquintino) [reply plans/DONE.md]: Fixed in f27bb4c — changed to `~544 lines` (approximate to avoid drift).

File: plans/DONE.md [hunk 4]
```diff
@@ -1,421 +1,152 @@
...
-- Documentation: `docs/src/guide/xarray-integration.md` — 7 worked examples covering the full conversion philosophy
-- Per-object metadata stored in descriptor extra keys (accessible via `desc.params`)
+- CLI Reference (all subcommands)
+- GRIB conversion overview + MARS key mapping
+- xarray integration guide
+- Edge Cases and Internals reference
+
+## Golden Test Files
+
+5 canonical `.tgm` files in `tests/golden/`:
```
[the remark this answers is not in the document]
Comment (tlmquintino) [reply plans/DONE.md]: Fixed in f27bb4c — updated path to `crates/tensogram-core/tests/golden/`.

File: plans/TEST.md [hunk 5]
```diff
@@ -1,44 +1,80 @@
...
+- xarray `open_dataset()` → lazy load → partial decode on slice access
+
+## Golden Test Files
+
+5 canonical `.tgm` files in `tests/golden/`:
+- `simple_f32.tgm` — single float32 object
+- `multi_object.tgm` — 3 dtypes: u8, i32, f64
+- `mars_metadata.tgm` — MARS keys in common/payload
+- `multi_message.tgm` — 3 independent messages
+- `hash_xxh3.tgm` — hash verification
```
[the remark this answers is not in the document]
Comment (tlmquintino) [reply plans/TEST.md]: Fixed in f27bb4c — updated path to `crates/tensogram-core/tests/golden/`.
